diff --git a/conf/machine/agilex_n6000.conf b/conf/machine/agilex_n6000.conf
new file mode 100644
index 0000000..1e14ab7
--- /dev/null
+++ b/conf/machine/agilex_n6000.conf
@@ -0,0 +1,35 @@
+#@TYPE: Machine
+#@NAME: Intel N6000
+#@DESCRIPTION: Machine configuration for the Agilex N6000
+
+require conf/machine/include/socfpga_armv8a.inc
+
+KBUILD_DEFCONFIG ?= "defconfig"
+
+UBOOT_CONFIG ??= "agilex_n6000"
+UBOOT_CONFIG[agilex_n6000] = "socfpga_agilex_n6000_defconfig"
+
+UBOOT_BINARY = "u-boot.itb"
+UBOOT_SUFFIX = "itb"
+
+KERNEL_CLASSES:append = " kernel-fitimage"
+KERNEL_IMAGETYPE = "Image"
+
+KERNEL_DEVICETREE ?= "intel/socfpga_agilex_n6000.dtb"
+
+
+UBOOT_EXTLINUX ?= "1"
+UBOOT_EXTLINUX_LABELS ?= "default"
+UBOOT_EXTLINUX_DEFAULT_LABEL ?= "Agilex N6000"
+
+UBOOT_EXTLINUX_MENU_DESCRIPTION:default ?= "Agilex N6000"
+UBOOT_EXTLINUX_KERNEL_IMAGE:default ?= "../${KERNEL_IMAGETYPE}"
+UBOOT_EXTLINUX_FDTDIR:default ?= ".."
+UBOOT_EXTLINUX_ROOT:default ?= "root=/dev/null"
+
+IMAGE_BOOT_FILES ?= " \
+			${UBOOT_BINARY} \
+			Image \
+			socfpga_agilex_n6000.dtb \
+		    "
+
diff --git a/recipes-bsp/arm-trusted-firmware/arm-trusted-firmware.inc b/recipes-bsp/arm-trusted-firmware/arm-trusted-firmware.inc
index 3e73261..64af89d 100644
--- a/recipes-bsp/arm-trusted-firmware/arm-trusted-firmware.inc
+++ b/recipes-bsp/arm-trusted-firmware/arm-trusted-firmware.inc
@@ -19,7 +19,7 @@ SRC_URI = "${ATF_REPO};protocol=${ATF_PROT};branch=${ATF_BRANCH}"
 
 S = "${WORKDIR}/git"
 
-COMPATIBLE_MACHINE = "(agilex|stratix10|n5x)"
+COMPATIBLE_MACHINE = "(agilex|agilex_n6000|stratix10|n5x)"
 ATFPLAT = "${MACHINE}"
 
 # Let the Makefile handle setting up the CFLAGS and LDFLAGS as it is a standalone application
diff --git a/recipes-bsp/u-boot/u-boot-socfpga-common.inc b/recipes-bsp/u-boot/u-boot-socfpga-common.inc
index 97b2715..5541427 100644
--- a/recipes-bsp/u-boot/u-boot-socfpga-common.inc
+++ b/recipes-bsp/u-boot/u-boot-socfpga-common.inc
@@ -4,6 +4,8 @@ SECTION = "bootloaders"
 DEPENDS = "coreutils-native bc-native bison-native dtc-native flex-native"
 DEPENDS:append = " python3-setuptools-native swig-native u-boot-mkimage-native"
 DEPENDS:append = " u-boot-tools virtual/kernel"
+DEPENDS:append:agilex_n6000 = " arm-trusted-firmware bash"
+COMPILE_PREPEND_FILES:agilex_n6000 = " bl31.bin Image linux.dtb rootfs.cpio.gz "
 
 LICENSE = "GPL-2.0-or-later"
 LIC_FILES_CHKSUM ?= "file://Licenses/README;md5=30503fd321432fc713238f582193b78e"
@@ -36,11 +38,19 @@ do_compile:prepend() {
 					if [ $j -eq $i ]; then
 						for file in ${COMPILE_PREPEND_FILES}; do
 							if [ "${file}" = "linux.dtb" ]; then
-								cp ${DEPLOY_DIR_IMAGE}/socfpga_${MACHINE}_socdk.dtb ${B}/${config}/linux.dtb
-								cp ${DEPLOY_DIR_IMAGE}/socfpga_${MACHINE}_socdk.dtb ${S}/linux.dtb
+								if [ "${MACHINE}" = "agilex_n6000" ]; then
+									SRC_DTB=${DEPLOY_DIR_IMAGE}/socfpga_agilex_n6000.dtb
+								else
+									SRC_DTB=${DEPLOY_DIR_IMAGE}/socfpga_${MACHINE}_socdk.dtb
+								fi
+								cp ${SRC_DTB} ${B}/${config}/linux.dtb
+								cp ${SRC_DTB} ${S}/linux.dtb
 							elif [ "${file}" = "u-boot.txt" ]; then
 								cp ${DEPLOY_DIR_IMAGE}/u-boot.txt ${B}/${config}/u-boot.txt
 								cp ${DEPLOY_DIR_IMAGE}/u-boot.txt ${S}/u-boot.txt
+							elif [ "${file}" = "rootfs.cpio.gz" ]; then
+								cp ${DEPLOY_DIR_IMAGE}/n6000-image-minimal-${MACHINE}.cpio.gz ${B}/${config}/${file}
+								cp ${DEPLOY_DIR_IMAGE}/n6000-image-minimal-${MACHINE}.cpio.gz ${S}/${file}
 							else
 								cp ${DEPLOY_DIR_IMAGE}/${file} ${B}/${config}/${file}
 								cp ${DEPLOY_DIR_IMAGE}/${file} ${S}/${file}
diff --git a/recipes-bsp/u-boot/u-boot-socfpga_v2023.04.bb b/recipes-bsp/u-boot/u-boot-socfpga_v2023.04.bb
new file mode 100755
index 0000000..a1c09cd
--- /dev/null
+++ b/recipes-bsp/u-boot/u-boot-socfpga_v2023.04.bb
@@ -0,0 +1,21 @@
+require u-boot-socfpga-common.inc
+
+DEPENDS:append:agilex = " arm-trusted-firmware bash"
+DEPENDS:append:stratix10 = " arm-trusted-firmware bash"
+DEPENDS:append:n5x = " arm-trusted-firmware bash"
+
+COMPILE_PREPEND_FILES:stratix10 = " bl31.bin "
+COMPILE_PREPEND_FILES:agilex = " bl31.bin "
+COMPILE_PREPEND_FILES:agilex_n6000 = " bl31.bin Image linux.dtb rootfs.cpio.gz "
+COMPILE_PREPEND_FILES:n5x = " bl31.bin Image linux.dtb u-boot.txt "
+COMPILE_PREPEND_FILES:arria10 = " Image "
+
+DEPLOY_APPEND_FILES:agilex = ""
+DEPLOY_APPEND_FILES:stratix10 = ""
+DEPLOY_APPEND_FILES:n5x = " kernel.itb "
+
+LIC_FILES_CHKSUM = "file://Licenses/README;md5=5a7450c57ffe5ae63fd732446b988025"
+
+UBOOT_VERSION = "v2023.04"
+
+SRCREV = "df8159c1e704dc445abdabc1dbe9484db867f5ba"
diff --git a/recipes-kernel/linux/config/nos10svc.cfg b/recipes-kernel/linux/config/nos10svc.cfg
new file mode 100644
index 0000000..635abcd
--- /dev/null
+++ b/recipes-kernel/linux/config/nos10svc.cfg
@@ -0,0 +1,2 @@
+CONFIG_INTEL_STRATIX10_SERVICE=n
+CONFIG_INTEL_STRATIX10_RSU=n
diff --git a/recipes-kernel/linux/config/nos10svc.scc b/recipes-kernel/linux/config/nos10svc.scc
new file mode 100644
index 0000000..cb411b8
--- /dev/null
+++ b/recipes-kernel/linux/config/nos10svc.scc
@@ -0,0 +1 @@
+kconf non-hardware nos10svc.cfg
diff --git a/recipes-kernel/linux/config/ppp.cfg b/recipes-kernel/linux/config/ppp.cfg
new file mode 100644
index 0000000..94c3299
--- /dev/null
+++ b/recipes-kernel/linux/config/ppp.cfg
@@ -0,0 +1,9 @@
+CONFIG_PPP=m
+CONFIG_PPP_BSDCOMP=m
+CONFIG_PPP_DEFLATE=m
+CONFIG_PPP_FILTER=y
+CONFIG_PPP_MPPE=m
+CONFIG_PPP_MULTILINK=y
+CONFIG_PPPOE=m
+CONFIG_PPP_ASYNC=m
+CONFIG_PPP_SYNC_TTY=m
diff --git a/recipes-kernel/linux/config/ppp.scc b/recipes-kernel/linux/config/ppp.scc
new file mode 100644
index 0000000..6008686
--- /dev/null
+++ b/recipes-kernel/linux/config/ppp.scc
@@ -0,0 +1 @@
+kconf non-hardware ppp.cfg
diff --git a/recipes-kernel/linux/config/uio.cfg b/recipes-kernel/linux/config/uio.cfg
new file mode 100644
index 0000000..769f6ca
--- /dev/null
+++ b/recipes-kernel/linux/config/uio.cfg
@@ -0,0 +1,2 @@
+CONFIG_UIO=m
+CONFIG_UIO_PDRV_GENIRQ=m
diff --git a/recipes-kernel/linux/config/uio.scc b/recipes-kernel/linux/config/uio.scc
new file mode 100644
index 0000000..7d606c3
--- /dev/null
+++ b/recipes-kernel/linux/config/uio.scc
@@ -0,0 +1 @@
+kconf non-hardware uio.cfg
diff --git a/recipes-kernel/linux/linux-socfpga-lts_6.1.bb b/recipes-kernel/linux/linux-socfpga-lts_6.1.bb
new file mode 100644
index 0000000..595f4b0
--- /dev/null
+++ b/recipes-kernel/linux/linux-socfpga-lts_6.1.bb
@@ -0,0 +1,8 @@
+LINUX_VERSION = "6.1.20"
+LINUX_VERSION_SUFFIX = "-lts"
+
+LIC_FILES_CHKSUM = "file://COPYING;md5=6bc538ed5bd9a7fc9398086aedcd7e46"
+
+SRCREV = "9cdbb1439bdf54faf242840b3ca388e7ef02244c"
+
+include linux-socfpga.inc
diff --git a/recipes-kernel/linux/linux-socfpga.inc b/recipes-kernel/linux/linux-socfpga.inc
index e286d06..3e87df4 100644
--- a/recipes-kernel/linux/linux-socfpga.inc
+++ b/recipes-kernel/linux/linux-socfpga.inc
@@ -26,3 +26,6 @@ require recipes-kernel/linux/linux-yocto.inc
 
 KBUILD_DEFCONFIG ?= "socfpga_defconfig"
 KCONFIG_MODE ?= "--alldefconfig"
+
+FILESEXTRAPATHS:prepend := "${THISDIR}/config:"
+SRC_URI:append:agilex_n6000 = " file://ppp.scc file://uio.scc file://nos10svc.scc "
