diff --git a/conf/machine/agilex7_n6000.conf b/conf/machine/agilex7_n6000.conf
new file mode 100644
index 0000000..d036f08
--- /dev/null
+++ b/conf/machine/agilex7_n6000.conf
@@ -0,0 +1,36 @@
+#@TYPE: Machine
+#@NAME: Intel N6000
+#@DESCRIPTION: Machine configuration for the Agilex7 N600
+
+require conf/machine/include/socfpga_armv8a.inc
+require conf/machine/include/ubifs.inc
+
+KBUILD_DEFCONFIG ?= "defconfig"
+
+UBOOT_CONFIG ??= "agilex7_n6000"
+UBOOT_CONFIG[agilex7_n6000] = "socfpga_agilex_n6000_defconfig"
+
+UBOOT_BINARY = "u-boot.itb"
+UBOOT_SUFFIX = "itb"
+
+KERNEL_CLASSES:append = " kernel-fitimage"
+KERNEL_IMAGETYPE = "Image"
+
+KERNEL_DEVICETREE ?= "intel/socfpga_agilex_n6000.dtb"
+
+
+UBOOT_EXTLINUX ?= "1"
+UBOOT_EXTLINUX_LABELS ?= "default"
+UBOOT_EXTLINUX_DEFAULT_LABEL ?= "Agilex N6000"
+
+UBOOT_EXTLINUX_MENU_DESCRIPTION:default ?= "Agilex N6000"
+UBOOT_EXTLINUX_KERNEL_IMAGE:default ?= "../${KERNEL_IMAGETYPE}"
+UBOOT_EXTLINUX_FDTDIR:default ?= ".."
+UBOOT_EXTLINUX_ROOT:default ?= "root=/dev/null"
+
+IMAGE_BOOT_FILES ?= " \
+			${UBOOT_BINARY} \
+			Image \
+			socfpga_agilex_n6000.dtb \
+		    "
+
diff --git a/recipes-bsp/arm-trusted-firmware/arm-trusted-firmware.inc b/recipes-bsp/arm-trusted-firmware/arm-trusted-firmware.inc
index 4c51c27..0a8cdf2 100644
--- a/recipes-bsp/arm-trusted-firmware/arm-trusted-firmware.inc
+++ b/recipes-bsp/arm-trusted-firmware/arm-trusted-firmware.inc
@@ -19,13 +19,14 @@ SRC_URI = "${ATF_REPO};protocol=${ATF_PROT};branch=${ATF_BRANCH}"
 
 S = "${WORKDIR}/git"
 
-COMPATIBLE_MACHINE = "(agilex7_dk_si_agf014ea|agilex7_dk_si_agf014eb|agilex7_dk_si_agi027fb|agilex7_dk_si_agi027fa|agilex7_dk_dev_agf027f1es|agilex5|stratix10|n5x)"
+COMPATIBLE_MACHINE = "(agilex7_dk_si_agf014ea|agilex7_dk_si_agf014eb|agilex7_dk_si_agi027fb|agilex7_dk_si_agi027fa|agilex7_dk_dev_agf027f1es|agilex7_n6000|agilex5|stratix10|n5x)"
 ATFPLAT = "${MACHINE}"
 ATFPLAT:agilex7_dk_si_agf014ea = "agilex"
 ATFPLAT:agilex7_dk_si_agf014eb = "agilex"
 ATFPLAT:agilex7_dk_si_agi027fb = "agilex"
 ATFPLAT:agilex7_dk_si_agi027fa = "agilex"
 ATFPLAT:agilex7_dk_dev_agf027f1es = "agilex"
+ATFPLAT:agilex7_n6000 = "agilex"
 
 LDFLAGS="--no-warn-rwx-segments"
 CFLAGS="-Wno-error=array-bounds"
diff --git a/recipes-bsp/u-boot/u-boot-socfpga-common.inc b/recipes-bsp/u-boot/u-boot-socfpga-common.inc
index 4171181..4fc6d7f 100644
--- a/recipes-bsp/u-boot/u-boot-socfpga-common.inc
+++ b/recipes-bsp/u-boot/u-boot-socfpga-common.inc
@@ -10,6 +10,7 @@ DEPENDS:append:agilex7_dk_si_agf014eb = " arm-trusted-firmware bash"
 DEPENDS:append:agilex7_dk_si_agi027fb = " arm-trusted-firmware bash"
 DEPENDS:append:agilex7_dk_si_agi027fa = " arm-trusted-firmware bash"
 DEPENDS:append:agilex7_dk_dev_agf027f1es = " arm-trusted-firmware bash"
+DEPENDS:append:agilex7_n6000 = " arm-trusted-firmware bash"
 DEPENDS:append:agilex5 = " arm-trusted-firmware bash"
 DEPENDS:append:stratix10 = " arm-trusted-firmware bash"
 DEPENDS:append:n5x = " arm-trusted-firmware bash"
@@ -19,6 +20,7 @@ COMPILE_PREPEND_FILES:agilex7_dk_si_agf014eb = " bl31.bin "
 COMPILE_PREPEND_FILES:agilex7_dk_si_agi027fb = " bl31.bin "
 COMPILE_PREPEND_FILES:agilex7_dk_si_agi027fa = " bl31.bin "
 COMPILE_PREPEND_FILES:agilex7_dk_dev_agf027f1es = " bl31.bin "
+COMPILE_PREPEND_FILES:agilex7_n6000 = " bl31.bin Image linux.dtb rootfs.cpio.gz "
 COMPILE_PREPEND_FILES:agilex5 = " bl31.bin "
 COMPILE_PREPEND_FILES:stratix10 = " bl31.bin "
 COMPILE_PREPEND_FILES:n5x = " bl31.bin Image linux.dtb u-boot.txt "
@@ -29,6 +31,7 @@ DEPLOY_APPEND_FILES:agilex7_dk_si_agf014eb = ""
 DEPLOY_APPEND_FILES:agilex7_dk_si_agi027fb = ""
 DEPLOY_APPEND_FILES:agilex7_dk_si_agi027fa = ""
 DEPLOY_APPEND_FILES:agilex7_dk_dev_agf027f1es = ""
+DEPLOY_APPEND_FILES:agilex7_n6000 = ""
 DEPLOY_APPEND_FILES:agilex5 = ""
 DEPLOY_APPEND_FILES:stratix10 = ""
 DEPLOY_APPEND_FILES:n5x = " kernel.itb "
@@ -64,11 +67,19 @@ do_compile:prepend() {
 					if [ $j -eq $i ]; then
 						for file in ${COMPILE_PREPEND_FILES}; do
 							if [ "${file}" = "linux.dtb" ]; then
-								cp ${DEPLOY_DIR_IMAGE}/socfpga_${MACHINE}_socdk.dtb ${B}/${config}/linux.dtb
-								cp ${DEPLOY_DIR_IMAGE}/socfpga_${MACHINE}_socdk.dtb ${S}/linux.dtb
+								if [ "${MACHINE}" = "agilex7_n6000" ]; then
+									SRC_DTB=${DEPLOY_DIR_IMAGE}/socfpga_agilex_n6000.dtb
+								else
+									SRC_DTB=${DEPLOY_DIR_IMAGE}/socfpga_${MACHINE}_socdk.dtb
+								fi
+								cp ${SRC_DTB} ${B}/${config}/linux.dtb
+								cp ${SRC_DTB} ${S}/linux.dtb
 							elif [ "${file}" = "u-boot.txt" ]; then
 								cp ${DEPLOY_DIR_IMAGE}/u-boot.txt ${B}/${config}/u-boot.txt
 								cp ${DEPLOY_DIR_IMAGE}/u-boot.txt ${S}/u-boot.txt
+							elif [ "${file}" = "rootfs.cpio.gz" ]; then
+								cp ${DEPLOY_DIR_IMAGE}/n6000-image-minimal-${MACHINE}.cpio.gz ${B}/${config}/${file}
+								cp ${DEPLOY_DIR_IMAGE}/n6000-image-minimal-${MACHINE}.cpio.gz ${S}/${file}
 							else
 								cp ${DEPLOY_DIR_IMAGE}/${file} ${B}/${config}/${file}
 								cp ${DEPLOY_DIR_IMAGE}/${file} ${S}/${file}
diff --git a/recipes-kernel/linux/config/ppp.cfg b/recipes-kernel/linux/config/ppp.cfg
new file mode 100644
index 0000000..94c3299
--- /dev/null
+++ b/recipes-kernel/linux/config/ppp.cfg
@@ -0,0 +1,9 @@
+CONFIG_PPP=m
+CONFIG_PPP_BSDCOMP=m
+CONFIG_PPP_DEFLATE=m
+CONFIG_PPP_FILTER=y
+CONFIG_PPP_MPPE=m
+CONFIG_PPP_MULTILINK=y
+CONFIG_PPPOE=m
+CONFIG_PPP_ASYNC=m
+CONFIG_PPP_SYNC_TTY=m
diff --git a/recipes-kernel/linux/config/ppp.scc b/recipes-kernel/linux/config/ppp.scc
new file mode 100644
index 0000000..6008686
--- /dev/null
+++ b/recipes-kernel/linux/config/ppp.scc
@@ -0,0 +1 @@
+kconf non-hardware ppp.cfg
diff --git a/recipes-kernel/linux/config/uio.cfg b/recipes-kernel/linux/config/uio.cfg
new file mode 100644
index 0000000..769f6ca
--- /dev/null
+++ b/recipes-kernel/linux/config/uio.cfg
@@ -0,0 +1,2 @@
+CONFIG_UIO=m
+CONFIG_UIO_PDRV_GENIRQ=m
diff --git a/recipes-kernel/linux/config/uio.scc b/recipes-kernel/linux/config/uio.scc
new file mode 100644
index 0000000..7d606c3
--- /dev/null
+++ b/recipes-kernel/linux/config/uio.scc
@@ -0,0 +1 @@
+kconf non-hardware uio.cfg
diff --git a/recipes-kernel/linux/linux-socfpga.inc b/recipes-kernel/linux/linux-socfpga.inc
index 96aacaa..9c13546 100644
--- a/recipes-kernel/linux/linux-socfpga.inc
+++ b/recipes-kernel/linux/linux-socfpga.inc
@@ -34,6 +34,7 @@ SRC_URI:append:agilex7_dk_si_agf014eb = " file://jffs2.scc file://gpio_sys.scc "
 SRC_URI:append:agilex7_dk_si_agi027fb = " file://jffs2.scc file://gpio_sys.scc "
 SRC_URI:append:agilex7_dk_si_agi027fa = " file://jffs2.scc file://gpio_sys.scc "
 SRC_URI:append:agilex7_dk_dev_agf027f1es = " file://jffs2.scc file://gpio_sys.scc "
+SRC_URI:append:agilex7_n6000 = " file://ppp.scc file://uio.scc "
 SRC_URI:append:agilex5 = " file://jffs2.scc file://gpio_sys.scc file://agilex5.scc"
 SRC_URI:append:stratix10 = " file://jffs2.scc file://gpio_sys.scc "
 SRC_URI:append:arria10 = " file://lbdaf.scc "
